DataFormats/PatCandidates/interface/PackedCandidate.h

```diff
@@ -664,13 +664,26 @@ namespace pat {
     reco::VertexRef::key_type pvRefKey_;
 
     /// IP covariance	
-    CMS_THREAD_GUARD(vertex_) mutable float dxydxy_, dzdz_, dxydz_,dlambdadz_,dphidxy_,dptdpt_,detadeta_,dphidphi_;
+    mutable std::atomic<reco::TrackBase::CovarianceMatrix *> m_;
     uint8_t packedHits_, packedLayers_; // packedLayers_ -> layers with valid hits; packedHits_ -> extra hits beyond the one-per-layer implied by packedLayers_
+    
     /// track quality information
     uint8_t normalizedChi2_; 
-//    uint8_t numberOfPixelHits_;
-  //  uint8_t numberOfHits_;
-    
+    uint16_t covarianceVersion_;
+    uint16_t covarianceSchema_;
+    CMS_THREAD_SAFE static CovarianceParameterization covarianceParameterization_;
+    //static std::atomic<CovarianceParameterization*> covarianceParameterization_;
+    static std::once_flag covariance_load_flag;
+    const CovarianceParameterization & covarianceParameterization() const {
+	std::call_once(covariance_load_flag,[](int v) { covarianceParameterization_.load(v); } ,covarianceVersion_ );
+        if(covarianceParameterization_.loadedVersion() != covarianceVersion_ )
+        {
+          std::cerr << "Attempting to load multiple covariance version in same process. This is not supported." << std::endl;
+	  abort();
```

Comment:
I think this was already discussed with Chris, the proper fix here is to handle multiple versions.. but that would require a proper mutex rather than std::call_once.